test/org/joml/test/AABBiTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2020 JOML.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.joml.test;
+
+import junit.framework.Assert;
+import junit.framework.TestCase;
+import org.joml.AABBi;
+
+public class AABBiTest extends TestCase {
+    public void testAABBiEdgeIntersection() {
+        AABBi center = new AABBi(0, 0, 0, 1, 1,1);
+        AABBi right = new AABBi(1, 0,0, 2, 1,1);
+        AABBi left = new AABBi(-1, 0, 0, 0, 1,1);
+        AABBi front = new AABBi(0, 1, 0, 1, 2,1);
+        AABBi back = new AABBi(0, -1, 0, 1, 0,1);
+        AABBi top = new AABBi(0, 0, 1, 1, 1,2);
+        AABBi bottom = new AABBi(0, 0, -1, 1, 1,0);
+
+
+        Assert.assertTrue(center.isValid());
+        Assert.assertTrue(right.isValid());
+        Assert.assertTrue(left.isValid());
+        Assert.assertTrue(top.isValid());
+        Assert.assertTrue(bottom.isValid());
+        Assert.assertTrue(front.isValid());
+        Assert.assertTrue(back.isValid());
+
+        // check contains
+        Assert.assertFalse(center.containsAABB(right));
+        Assert.assertFalse(center.containsAABB(left));
+        Assert.assertFalse(center.containsAABB(front));
+        Assert.assertFalse(center.containsAABB(back));
+        Assert.assertFalse(center.containsAABB(top));
+
+        // test intersection
+        Assert.assertTrue(center.intersectsAABB(right));
+        Assert.assertTrue(center.intersectsAABB(left));
+        Assert.assertTrue(center.intersectsAABB(front));
+        Assert.assertTrue(center.intersectsAABB(back));
+        Assert.assertTrue(center.intersectsAABB(top));
+
+        Assert.assertTrue(right.intersectsAABB(center));
+        Assert.assertTrue(left.intersectsAABB(center));
+        Assert.assertTrue(front.intersectsAABB(center));
+        Assert.assertTrue(back.intersectsAABB(center));
+        Assert.assertTrue(top.intersectsAABB(center));
```

Comment:
I'd say these are actually multiple tests (for different properties). I would structure this a bit different, instantiating the AABBs in a `@BeforeAll` method and then have dedicated test methods for the respective properties, e.g.,

```java
public void testAABBiValidity() {
        Assert.assertTrue(center.isValid());
        Assert.assertTrue(right.isValid());
        Assert.assertTrue(left.isValid());
        Assert.assertTrue(top.isValid());
        Assert.assertTrue(bottom.isValid());
        Assert.assertTrue(front.isValid());
        Assert.assertTrue(back.isValid());
}
```

Might even use parameterized tests or fancy stuff like that, but even splitting this up manually should make it a bit cleaner...